instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/TracingExecutionInterceptor.java

```diff
@@ -289,6 +303,43 @@ private void onSdkResponse(
     }
   }
 
+  @Override
+  public void afterTransmission(
+      Context.AfterTransmission context, ExecutionAttributes executionAttributes) {
+    if (recordIndividualHttpError) {
+      extractHttpErrorAsEvent(context, executionAttributes);
+    }
+  }
+
+  private static void extractHttpErrorAsEvent(
+      Context.AfterTransmission context, ExecutionAttributes executionAttributes) {
+    io.opentelemetry.context.Context otelContext = getContext(executionAttributes);
+    if (otelContext != null) {
+      Span span = Span.fromContext(otelContext);
+      SdkHttpResponse response = context.httpResponse();
+
+      if (!response.isSuccessful()) {
```

Comment:
Can response ever be null on the context? Should a null check be added?

Reply:
Make sense. I will add a null check on response.